### PR TITLE
Implementation For Yield Offering Related APIs

### PR DESCRIPTION
**Pull request by leonardchinonso**
##Implementation
- Added models for all yield related data objects
- Added methods in the RemoteCall interface to facilitate yield related requests
- Added an example package to test remote calls

##Paired With
CC @odetolakehinde

### CODE REVIEW

File: api/yield.go [hunk 1]
```diff
@@ -1 +1,177 @@
...
+		SetContext(ctx).
+		Get(endpoint)
+
+	if err != nil {
+		fL.Err(err).Msg("error occurred")
+		return []model.YieldOfferingProfile{}, err
+	}
+
+	if res.StatusCode() != http.StatusOK {
+		fL.Info().Str("error_code", fmt.Sprintf("%d", res.StatusCode())).Msg(string(res.Body()))
```
Comment (odetolakehinde) [inline api/yield.go:139]: can we have the `error_code` to be in constants? `model.LogErrorCode`

**Review by odetolakehinde — APPROVED** [review]

**Comment by odetolakehinde** [comment]
@wharleyinc @azeezlala

**Review by wharleyinc — COMMENTED** [review]
@odetolakehinde looks good to me

File: api/yield.go [hunk 2]
```diff
@@ -1 +1,177 @@
...
+	fL.Info().Interface(model.LogStrRequest, "empty").Msg("request")
+	defer fL.Info().Msg("done...")
+
+	response := struct {
+		Data []model.Portfolio `json:"data"`
+	}{}
+	res, err := c.client.R().
+		SetAuthToken(c.bearerToken).
+		SetResult(&response).
+		SetHeader("X-Idempotent-ID", c.idempotentID.String()).
```
Comment (azeezlala) [inline api/yield.go:28]: No needed sha

**Review by azeezlala — CHANGES_REQUESTED** [review]

**Review by odetolakehinde — APPROVED** [review]

**Review by azeezlala — APPROVED** [review]
LGTM

**Pull request merged**